Hexastore/Graph/TripleObject.cs

```diff
@@ -182,7 +187,8 @@ public override bool Equals(object obj)
             //    return t.IsID == IsID && t.Value == Value && t.TokenType == TokenType;
             //}
 
-            return t.IsID == IsID && t.Value == Value && t.Index == Index && t.TokenType == TokenType;
+            // NOTE: the ! on !t.IsID == !IsID seems to be necessary as without it, it ends up comparing references and not values
+            return !t.IsID == !IsID && t.Value == Value && t.Index == Index && t.TokenType == TokenType;
```

Comment:
``return !t.IsID == !IsID && t.Index == Index && t.TokenType == TokenType && StringComparer.Ordinal(t.Value, Value);``

Unrelated to this change, but for perf reasons the string compare should be last and done with ordinal